drivers/SmartThings/bose/src/init.lua

```diff
@@ -147,71 +148,77 @@ local function do_refresh(driver, device, cmd)
   -- get presets
   local presets, err = command.presets(device:get_field("ip"))
   if not presets then
-    log.error(string.format("failed to get presets: %s", err))
+    device.log.error(string.format("failed to get presets: %s", err))
   else
     device:emit_event(capabilities.mediaPresets.presets(presets))
   end
-end
 
--- build a exponential backoff time value generator
---
--- max: the maximum wait interval (not including `rand factor`)
--- inc: the rate at which to exponentially back off
--- rand: a randomization range of (-rand, rand) to be added to each interval
-local function backoff_builder(max, inc, rand)
-  local count = 0
-  inc = inc or 1
-  return function()
-    local randval = 0
-    if rand then
-      -- random value in range (-rand, rand)
-      randval = math.random() * rand * 2 - rand
+  -- restart listener if needed
+  local listener = device:get_field("listener")
+  local success = false
+  if listener and (listener:is_stopped() or listener.websocket == nil)then
+    device.log.info("Restarting listening websocket client for device updates")
+    listener:stop()
```

Comment:
does this do anything if `is_stopped` is already true or the `websocket` is `nil`?